### ISSUE TITLE
Setup emp_main to wait rather than exit

### ISSUE THREAD

**Issue by mercere99** [devosoft/Empirical#192]
A major error in Empirical occurs because main goes out of scope and frees objects created there.  This change would fix that problem.

emp_main() behaves identical to main currently, except it waits until everything is loaded before starting.

Perhaps we should allow a signal to trigger emp_main to end in case that's needed at some point.

**Comment by mercere99** [comment]
In retrospect, I'm not sure how easy this modification will be; its challenging to stop at a close-brace without ending a scope.  I think the better way to fix this problem will be with an improved Template project.
